### ISSUE TITLE
Out of memory while scanning local library

### ISSUE THREAD

**Issue by hank** [mopidy/mopidy#1204]
```
(mopidy:10105): GLib-ERROR **: gmem.c:103: failed to allocate 4294967286 bytes
```
When running local scan with sqlite extension.

**Comment by jodal** [comment]
That's 4.3GB of memory. Sounds like an out-of-memory situation.

**Comment by hank** [comment]
Any idea whether this would be in your ballpark or whoever wrote the sqlite extension?

**Comment by jodal** [comment]
GLib indicates GStreamer, so it's probably Mopidy core or an GStreamer/GLib issue. I suppose Mopidy-Local-Sqlite is innocent.

That said, there's not much to go on here. Can you reproduce this?

**Comment by tkem** [comment]
Since this occurs with a local scan, I guess the output of `mopidy -v local scan` would be helpful, too.

**Comment by jodal** [comment]
Not really anything to go on here, so I'm closing this. Anyone experiencing this issue, feel free to comment here so we get an idea of how frequent this issue surfaces.

**Comment by virtadpt** [comment]
I just started getting this while setting up Mopidy v1.1.1 on a RasPi model B running Raspbian, installed with the official instructions.  I'm running the command `mopidyctl local scan` with the Mopidy-Local-Sqlite extension installed on a collection of 9,687 files in my music collection (this is the initial setup run).

I've just run `mopidyctl -v local scan` and, debugging output aside, I still get the same message before it dies: ```(mopidy:7072): GLib-ERROR **: /usr/src/packages/BUILD/glib2.0-2.40.0/./glib/gmem.c:103: failed to allocate 4294967286 bytes```

**Comment by tkem** [comment]
@virtadpt: Please post the last 10 lines if debug output, so we might get an idea what led to this.

**Comment by virtadpt** [comment]
I figured out what the problem is - Mopidy runs out of memory and dies from an OOM error.  I've been able to fix this problem (on my RasPi, anyway) by moving all of my music into a temp directory, moving two or three hundred files back into the music directory, and rerunning `sudo mopidyctl local scan` each time (I'm using mopidy-local-sqlite as the back-end datastore).  It takes some work, but just as much time as running the local scan against the entire collection at once and having it die each time.

**Comment by virtadpt** [comment]
Some observations:

1.1 GB of files to scan is too much on a RasPi.

Occasionally, a strangely encoded .mp3 will cause the same thing to happen (I found one file in my library that consistently caused the problem - deleting it fixed the problem handily).

Scanning up to 500 files at a time on a RasPi seems to work fine.

**Comment by jodal** [comment]
Mopidy 2.0 was released yesterday. It upgrades GStreamer, which we use to scan the files, from 0.10 to 1.x. In other words, we now have several years of accumulated bug fixes in GStreamer available to us.

@virtadpt Can you check if Mopidy 2.0 is better at this?

**Comment by virtadpt** [comment]
I'll give it a try later this week, I just got off the road for work and I need to catch up on everything at the office.  I'll let you know how it works.

**Comment by jodal** [comment]
Closing due to inactivity. Probably fixed by Mopidy 2.0 upgrading to GStreamer 1.x.

**Comment by bigredfrog** [comment]
I just saw something very similar on Mopidy 2.0 GStreamer 1.0

Happens ever time, where can I get more diagnostic?

I know it didn't occur when I turned wma files off, did occur even though I didn't have a wma codec at the time if I tried to process them.


INFO     Scanned 1600 of 6912 files in 199s, ~661s left.

(python:10115): GLib-ERROR **: /build/glib2.0-NyxCB6/glib2.0-2.48.1/./glib/gmem.c:100: failed to allocate 3634425503 bytes
